ts/parsers/paragraphparser.ts

```diff
@@ -1,37 +1,155 @@
-import { getValueAtPath } from "../helpers";
+import { checkPath, getValueAtPath } from "../helpers";
 import { ColorParser } from "./";
 
 import {
     PowerpointElement,
     TextAlignment,
     FontAttributes,
     Paragraph,
-    Content
+    Content,
+    List,
+    ListType
 } from "airppt-models-plus/pptelement";
 
 /**
  * Parse the paragraph elements
  */
 export default class ParagraphParser {
+    public static isList(paragraph): boolean {
+        return (
+            checkPath(paragraph, '["a:pPr"][0]["a:buAutoNum"]') ||
+            checkPath(paragraph, '["a:pPr"][0]["a:buChar"]')
+        );
+    }
+
+    public static getParagraph(paragraph): Paragraph {
+        const textElements = paragraph["a:r"] || [];
+        const content = textElements.map((txtElement) => {
+            return {
+                text: txtElement["a:t"] || "",
+                textCharacterProperties: this.determineTextProperties(
+                    getValueAtPath(txtElement, '["a:rPr"][0]')
+                )
+            };
+        });
+
+        return {
+            content: content,
+            paragraphProperties: this.determineParagraphProperties(paragraph)
+        };
+    }
+
+    public static getListlevel(paragraph): number {
+        const level = getValueAtPath(paragraph, '["a:pPr"][0]["$"]["lvl"]');
+
+        return level ? parseInt(level) : 0;
```

Comment:
it feels a bit dangerous to assume level 0 for the list items where level cannot be deciphered from XML. Wouldn't it be better to ensure the list is at a valid and known level inside `isList` function and drop items at unknown levels completely?

Unless there is a specific use case for all level 0 items or something .. I wouldn't put it beyond MS tbh.

Reply:
the function is being only called only for the list item. 
MS doesn't give you the level 0 for the top level list item. therefore returning 0 for it. 
`lvl` is there only from level 1 and so on

Reply:
> Unless there is a specific use case for all level 0 items or something .. I wouldn't put it beyond MS tbh.

> MS doesn't give you the level 0 for the top level list item

I knew they will do some crazy shit like that .... ah MS. 

OK, fine,